jupyter_client/manager.py

```diff
@@ -454,6 +454,19 @@ def is_alive(self):
             # we don't have a kernel
             return False
 
+    def memory_usage(self):
+        """
+        Returns memory usage of the kernel in MB.
+        Function should be overwritten if there is a
+        need for a different kind of ram check, hence the import of the
+        'psutil' package is inside the memory usage function
+        """
+        from psutil import Process
+        kernel_pid = self.kernel.pid
+        process = Process(kernel_pid)
+        mem = process.memory_info().vms  # in bytes
+        return int(mem / 1024**2)  # in MB
+
```

Comment:
I'm definitely cool with this for a start.

Reply:
I'm inclined to have this be more generic and have it return a json or dict where `memory_usage` is the only field (for now).  This way there's room for growth.  Of course, the consumer of this method would need to know how to interpret fields - but that's the beauty of json/dicts (they can decide what they want to interpret).  

I could definitely see this as a method on a kernel provider in the kernel-mgmt stuff that @takluyver is floating.  So using something like `kernel_stats()` may be better suited down the road.  Just a thought.

Reply:
@kevin-bates 

you said, we can make the method return a json containing different types of infos about the kennel.
The question is, what if the user wants to get a very specific type of info, for example only the memory usage, and dosent want to wait for all of the other data to be gathered?

Would you say it will be wise to break down the gathering of the info to different methods and then add another option which is a method that takes all of the info that we are able to get and put it into an easy to use json.

What I mean is, for example a method called `_get_memory_usage` and other methods for different options, and then another method called `kernel_stats` that will return a json as you said with the info in it.